K8SSAND-1071 ⁃ Cassandra data dir fails to be created in 1.4.0

## Bug Report
<!--
Thanks for filing an issue! Before hitting the button, please answer these questions.
Fill in as much of the template below as you can. 
-->

**Describe the bug**
When deploying K8ssandra 1.4.0 on GKE, Cassandra pods fail to start properly due to the following error:

```
Caused by: java.lang.IllegalStateException: Failed to mkdirs /opt/cassandra/data/data; unable to start server
	at org.apache.cassandra.config.DatabaseDescriptor.createAllDirectories(DatabaseDescriptor.java:1422)
	at org.apache.cassandra.db.commitlog.CommitLog.<init>(CommitLog.java:94)
	at org.apache.cassandra.db.commitlog.CommitLog.construct(CommitLog.java:77)
	at org.apache.cassandra.db.commitlog.CommitLog.<clinit>(CommitLog.java:64)
	... 3 common frames omitted
```

This is apparently due to newly added pod level `securityContext` entries. 
Removing those fixes the issue and Cassandra starts properly.

* Kubernetes cluster kind:
GKE



┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1071) by [Unito](https://www.unito.io)
┆Reviewer: 5e66ca0cae59460cfb01a13c
┆fixVersions: k8ssndra-v1.4.1
┆friendlyId: K8SSAND-1071
┆priority: Medium

--- a/CHANGELOG-1.4.md
+++ b/CHANGELOG-1.4.md
@@ -16,6 +16,9 @@ last).
 When cutting a new release of the parent `k8ssandra` chart update the `unreleased` heading to the tag being generated 
 and date `## vX.Y.Z - YYYY-MM-DD` and create a new placeholder section for  `unreleased` entries.
 
+## unreleased
+* [BUGFIX] Remove pod level SecurityContext to fix permissions issue on Cassandra data dir creation
+
 ## v1.4.0 - 2021-11-19
 * [CHANGE] Update to Managment API v0.1.33
 * [CHANGE] Update to Medusa 0.11.3

--- a/charts/k8ssandra/templates/cassandra/cassdc.yaml
+++ b/charts/k8ssandra/templates/cassandra/cassdc.yaml
@@ -198,7 +198,6 @@ spec:
 {{- end }}
   podTemplateSpec:
     spec:
-      securityContext: {{- toYaml .Values.cassandra.podSecurityContext | nindent 8 }}
       initContainers:
       - name: base-config-init
         image: {{ include "k8ssandra.cassandraImage" . }}

--- a/charts/k8ssandra/values.yaml
+++ b/charts/k8ssandra/values.yaml
@@ -42,8 +42,6 @@ cassandra:
   # -- Security context override for cassandra container
   securityContext: {}
 
-  # -- Security context override for pod where Cassandra container resides
-  podSecurityContext: {}
   # -- Cassandra base init container
   baseConfig:
     # -- Security context override for base init container

